src/functionaltests/java/com/ericsson/ei/subscriptions/crud/SubscriptionCRUDSteps.java

```diff
@@ -34,87 +32,105 @@
+        String found = node.get("foundSubscriptions").get(0).toString();
+        Subscription subscription = mapper.readValue(found, Subscription.class);
         assertEquals(notificationType, subscription.getNotificationType());
     }
-    //Scenario:3 ends ==========================================================================================
+    // Scenario:3 ends ==================================================================
 
     @When("^I make a DELETE request with subscription name \"([^\"]*)\" to the  subscription REST API \"([^\"]*)\"$")
-    public void i_make_a_DELETE_request_with_subscription_name_to_the_subscription_REST_API(String name, String endPoint) throws Throwable {
-        result = mockMvc.perform(MockMvcRequestBuilders.delete(endPoint + name).accept(MediaType.APPLICATION_JSON)).andReturn();
-    }
-
-    @Then("^I get response code of (\\d+) for successful delete$")
-    public void i_get_response_code_of_for_successful_delete(int statusCode) throws Throwable {
-        Assert.assertEquals(statusCode, result.getResponse().getStatus());
+    public void i_make_a_DELETE_request_with_subscription_name_to_the_subscription_REST_API(String name,
+            String endPoint) throws Throwable {
+        httpRequest = new HttpRequest(HttpMethod.DELETE);
+        httpRequest.setHost(hostName).setPort(applicationPort).setEndpoint(endPoint + name)
+                .addHeader("Accept", "application/json");
+        response = httpRequest.performRequest();
     }
 
-    @Then("^My GET request with subscription name \"([^\"]*)\" at REST API \"([^\"]*)\" returns empty String \"([^\"]*)\"$")
-    public void my_GET_request_with_subscription_name_at_REST_API_returns_empty_String(String name, String endPoint, String emptyString) throws Throwable {
-        result = mockMvc.perform(MockMvcRequestBuilders.get(endPoint + name)
-            .accept(MediaType.APPLICATION_JSON))
-            .andReturn();
-        JSONObject response = new JSONObject(result.getResponse().getContentAsString());
-        assertEquals(emptyString, response.getJSONArray("foundSubscriptions").toString());
-        assertEquals(name, response.getJSONArray("notFoundSubscriptions").getString(0));
+    @Then("^My GET request with subscription name \"([^\"]*)\" at REST API \"([^\"]*)\" returns an empty String$")
+    public void my_GET_request_with_subscription_name_at_REST_API_returns_empty_String(String name, String endPoint)
+            throws Throwable {
+        httpRequest = new HttpRequest(HttpMethod.GET);
+        httpRequest.setHost(hostName).setPort(applicationPort).setEndpoint(endPoint + name)
+                .addHeader("Accept", "application/json");
+        response = httpRequest.performRequest();
+        JsonNode node = eventManager.getJSONFromString(response.getBody());
+        int size = node.get("foundSubscriptions").size();
+        String notFound = node.get("notFoundSubscriptions").get(0).asText();
+        assertEquals(0, size);
```

Comment:
Since you are only comparing numbers, it will be hard to simple see what it failed on. Add a message regarding what failed to the assertEquals: e.g. assertEquals(<message>, expected, actual)